src/process_utility.c

```diff
@@ -882,7 +914,142 @@ timescaledb_ProcessUtility(Node *parsetree,
+{
+	switch (nodeTag(cmd->parsetree))
+	{
+		case T_AlterTableStmt:
+			process_altertable(cmd->parsetree, cmd);
+			break;
+		case T_CreateTrigStmt:
+			process_create_trigger(cmd->parsetree);
+			break;
+		default:
+			break;
+	}
+}
+
+/*
+ * ProcessUtility hook for DDL commands that have not yet been processed by
+ * PostgreSQL.
+ */
+static void
+timescaledb_ddl_command_start(Node *parsetree,
+							  const char *query_string,
+							  ProcessUtilityContext context,
+							  ParamListInfo params,
+							  DestReceiver *dest,
+							  char *completion_tag)
+{
+	/*
+	 * If we are restoring, we don't want to recurse to chunks or block
+	 * operations on chunks. If we do, the restore will fail.
+	 */
+	if (!extension_is_loaded() || guc_restoring)
+	{
+		prev_ProcessUtility(parsetree, query_string, context, params, dest, completion_tag);
+		return;
+	}
+
+	if (!process_ddl_command_start(parsetree, query_string, context, completion_tag))
+		prev_ProcessUtility(parsetree, query_string, context, params, dest, completion_tag);
+}
+
+/* Function manager info for the event "pg_event_trigger_ddl_commands", which is
+ * used to retrieve information on executed DDL commands in an event
+ * trigger. The function manager info is initialized on extension load. */
+static FmgrInfo ddl_commands_fmgrinfo;
+
+/*
+ * Get a list of executed DDL commands in an event trigger.
+ *
+ * This function calls the function pg_event_trigger_ddl_commands(), which is
+ * part of the event trigger API, and retrieves the DDL commands executed in
+ * relation to the event trigger. It is only valid to call this function from
+ * within an event trigger.
+ */
+static List *
+event_trigger_ddl_commands()
+{
+	ReturnSetInfo rsinfo;
+	FunctionCallInfoData fcinfo;
+	TupleTableSlot *slot;
+	EState	   *estate = CreateExecutorState();
+	List	   *objects = NIL;
+
+	InitFunctionCallInfoData(fcinfo, &ddl_commands_fmgrinfo, 1, InvalidOid, NULL, NULL);
+	MemSet(&rsinfo, 0, sizeof(rsinfo));
+	rsinfo.type = T_ReturnSetInfo;
+	rsinfo.allowedModes = SFRM_Materialize;
+	rsinfo.econtext = CreateExprContext(estate);
+	fcinfo.resultinfo = (fmNodePtr) &rsinfo;
+
+	FunctionCallInvoke(&fcinfo);
+
+	slot = MakeSingleTupleTableSlot(rsinfo.setDesc);
+
+	while (tuplestore_gettupleslot(rsinfo.setResult, true, false, slot))
+	{
+		HeapTuple	tuple = ExecFetchSlotTuple(slot);
+		CollectedCommand *cmd;
+		Datum		values[9];
+		bool		nulls[9];
+
+		heap_deform_tuple(tuple, rsinfo.setDesc, values, nulls);
+
+		if (!nulls[8])
+		{
+			cmd = (CollectedCommand *) DatumGetPointer(values[8]);
+			objects = lappend(objects, cmd);
+		}
+	}
+
+	FreeExprContext(rsinfo.econtext, false);
+	FreeExecutorState(estate);
+	ExecDropSingleTupleTableSlot(slot);
+
+	return objects;
+}
+
+PG_FUNCTION_INFO_V1(timescaledb_ddl_command_end);
+
+/*
+ * Event trigger hook for DDL commands that have alread been handled by
+ * PostgreSQL (i.e., "ddl_command_end" events).
+ */
+Datum
+timescaledb_ddl_command_end(PG_FUNCTION_ARGS)
```

Comment:
Should this function name make clear this is an event trigger?

Reply:
Figured it was clear from the comment and data types in the function. I preferred making the naming consistent with the process utility hook.